pages/_app.js

```diff
@@ -30,14 +30,33 @@ export default function App({ Component, pageProps }) {
     router.push(`/detail/book/${currentBook.id}/detail`);
   }
 
+  function newBook(currentBook) {
+    const newBook = {
+      ...currentBook,
+      id: String(bookCounter + 1),
```

Comment:
you should choose a different approach for creating ids for your books. Since your current logic will break as soon as you implement a feature to remove books.

You can work with the crypto module, for instance

Reply:
Change id to v4. I added an internalId for people who need their own label. Also, later I will not delete entries in a database, but only set them to inactive. Then you can reactivate accidentally deleted entries.